### ISSUE TITLE
Can it be used on Arduino IDE for ESP32 

### ISSUE THREAD

**Issue by ElcRcp** [richgel999/miniz#261]
Hello, firstly thanks for your work and share.
Currently I'm trying to figure out how to compress/zip data/file on a ESP32 microcontroller and wandering in google for it. Whenever I do search for esp32 compressing algorithms and libraries I see this work. I wasn't sure if this library will work with Arduino IDE but I tried to copy .c,.h files and test the example but wasn't able to success, so I wonder if this library works with Arduino IDE and if it is, how?

**Comment by WereCatf** [comment]
Seems to be working just fine. I disabled file I/O, time functions and the archive APIs in miniz.h, since those weren't needed for a quick test, then I just copied a portion of example1.c to a sketch, flashed the sketch to an ESP32-S2 and it compressed the sample string without an issue.

That said, saying "it doesn't work, how to make it work?" isn't useful; no one can help you with that little information.

**Comment by ElcRcp** [comment]
Thanks for the tips, I'll try again 
Actually what I mean with "it doesn't work, how to make it work?" ; Since I wasn't able to make it work so I asked if there is something I should do extra other then copying library files to project folder but I believe there isn't since you did not mention anything special.

**Comment by WereCatf** [comment]
@ElcRcp Just use the files from the release, not the raw files from the repository directly.

**Comment by ElcRcp** [comment]
Library working on ESP32-Wrover series. Using example_1 I was able to compress 470kB data to 96kB on PSRAM. Next I'll try file compression on FFAT or SPIFFS but I'm closing this issue since I got my answer for initial question.
As a last comment for whoever want to use this on an microcontroller, such like mz_uint/mz_int variable definitions are defined with int, short, double etc. in header file so I recommend changing them with corresponding int/uint_32/64/16/8 before using library.
Thanks for the job and help.
